app/ui/reports_query_hash.py

```diff
@@ -0,0 +1,158 @@
+            go.Pie(labels=labels, values=df.Cost.values, name="Cost per bucket"), 1, 1
+        )
+        fig.add_trace(
+            go.Pie(labels=labels, values=df.Count.values, name="Count per bucket"), 1, 2
+        )
+
+        fig.update_traces(hole=0.4, hoverinfo="label+percent+name")
+
+        fig.update_layout(
+            title_text="Repeated queries per frequency bucket",
+            # Add annotations in the center of the donut pies.
+            annotations=[
+                dict(text="Cost", x=0.20, y=0.5, font_size=20, showarrow=False),
+                dict(text="Count", x=0.81, y=0.5, font_size=20, showarrow=False),
+            ],
+        )
+        st.plotly_chart(fig, use_container_width=True)
+
+    def top_table():
+        sql = f"""
+            with raw as (
+            select any_value(query_text) as "Query Text", sum(cost) as "Cost", count(*) as "Count", query_parameterized_hash as "Parameterized Query Hash"
+            from reporting.labeled_query_history where query_parameterized_hash is not null and cost >0
+                and start_time between %(start)s and %(end)s and (array_size(%(warehouse_names)s) = 0
+                    OR array_contains(warehouse_name::variant, %(warehouse_names)s))
+                    {addition_filter}
+                    group by query_parameterized_hash
+                    order by "Count" desc
+                    ) select * from raw where length("Query Text") > 0
+                    limit 100
+                    """
+        df = connection.execute_select(
+            sql + ";",
+            {"start": bf.start, "end": bf.end, "warehouse_names": bf.warehouse_names},
+        )
+        st.markdown(
+            """
+        Top 100 repeated queries by count.
+
+        Tips:
+        * double click on the query text to see the full query
+        * copy the query hash and run `select * from opscenter.reporting.enriched_query_history where query_parameterized_hash = '<hash>'` to see the full query history for a particular hash
+        """
+        )
+        st.dataframe(df, use_container_width=True)
+
+    is_enabled = connection.execute_select(
+        """select system$BEHAVIOR_CHANGE_BUNDLE_STATUS('2023_06' = 'ENABLED' and count(*) = 1
+        from information_schema.columns where table_catalog='DRYAN' and table_name='ENRICHED_QUERY_HISTORY' and table_schema='REPORTING' and column_name='QUERY_HASH';"""
+    ).values[0][0]
+    if not is_enabled:
+        st.markdown(
+            """
+        **Note:** This report requires the `QUERY_HASH` column to be enabled in the `ENRICHED_QUERY_HISTORY` table.
+        This column is enabled by default in Snowflake 2023.06. If you are on an earlier version, you can enable it
+        by running the following command:
+
+        ```
+        BEGIN
+          SELECT SYSTEM$ENABLE_BEHAVIOR_CHANGE_BUNDLE('2023_06');
+          EXECUTE opscenter.tasks.query_history_maintenance;
+          CALL opscenter.admin.finalize_setup();
```

Comment:
nit: superfluous as finalize_setup is going to trigger the QH task

Reply:
thanks, fixed